spec/PeriodicBackgroundSync-index.bs

```diff
@@ -204,3 +205,9 @@ The <code><dfn method for=PeriodicSyncManager title="register(tag, options)">reg
         1. Set |currentRegistration|'s associated [=state=] to [=pending=].
     1. Else, if |serviceWorkerRegistration|'s [=registration state=] is [=firing=], set |serviceWorkerRegistration|'s [=registration state=] to [=reregisteredWhileFiring=].
     1. [=Resolve=] |promise|.
+
+The <code><dfn method for=PeriodicSyncManager title="getTags()">getTags()</dfn></code> method when invoked, MUST return [=a new promise=] |promise| and run the following steps [=in parallel=]:
```

Comment:
This is racy, a parallel queue will solve this. I linked to it in another review

Reply:
In parallel, which I'm suggesting here, links to the same concept:
https://html.spec.whatwg.org/multipage/infrastructure.html#in-parallel

Reply:
Yes, it proceeds to explain how to use parallel queues to get rid of race conditions (https://html.spec.whatwg.org/multipage/infrastructure.html#parallel-queue).

If you use the same parallel queue for all the algorithms you can also get rid of the re-registered while firing state.

Reply:
done.
Discussed offline why the reregisteredwhileFiring state is still necessary.